v2/datastream-to-spanner/terraform/samples/mysql-sharded-end-to-end/terraform.tfvars

```diff
@@ -0,0 +1,109 @@
+      datastream_source_type              = "<YOUR_DATASTREAM_SOURCE_TYPE>" # Datastream source type (e.g., "MYSQL")
+      round_json_decimals                 = "<TRUE/FALSE>"                  # Whether to round JSON decimal values in Dataflow
+      directory_watch_duration_in_minutes = "<YOUR_WATCH_DURATION>"         # Directory watch duration (in minutes) for Dataflow
+      spanner_priority                    = "<YOUR_SPANNER_PRIORITY>"       # Spanner priority ("high", "medium", or "low")
+      local_session_file_path             = "<YOUR_SESSION_FILE_PATH>"      # Path to local session file (optional)
+    }
+
+    runner_params = {
+      additional_experiments = ["enable_google_cloud_profiler", "enable_stackdriver_agent_metrics",
+      "disable_runner_v2", "enable_google_cloud_heap_sampling"]
+      autoscaling_algorithm        = "<YOUR_AUTOSCALING_ALGORITHM>"   # e.g., "THROUGHPUT_BASED", "NONE"
+      enable_streaming_engine      = "<TRUE/FALSE>"                   # Whether to use Dataflow Streaming Engine
+      kms_key_name                 = "<YOUR_KMS_KEY_NAME>"            # KMS key name for encryption (optional)
+      labels                       = { env = "<YOUR_ENVIRONMENT>" }   # Labels for the Dataflow job
+      launcher_machine_type        = "<YOUR_LAUNCHER_MACHINE_TYPE>"   # Machine type for the launcher VM (e.g., "n1-standard-1")
+      machine_type                 = "<YOUR_MACHINE_TYPE>"            # Machine type for worker VMs (e.g., "n2-standard-2")
+      max_workers                  = "<YOUR_MAX_WORKERS>"             # Maximum number of worker VMs
+      job_name                     = "<YOUR_JOB_NAME>"                # Name of the Dataflow job
+      network                      = "<YOUR_VPC_NETWORK>"             # VPC network for the Dataflow job
+      num_workers                  = "<YOUR_NUM_WORKERS>"             # Initial number of worker VMs
+      sdk_container_image          = "<YOUR_SDK_CONTAINER_IMAGE>"     # Dataflow SDK container image
+      service_account_email        = "<YOUR_SERVICE_ACCOUNT_EMAIL>"   # Service account email for Dataflow
+      skip_wait_on_job_termination = "<TRUE/FALSE>"                   # Whether to skip waiting for job termination on deletion
+      staging_location             = "gs://<YOUR_GCS_BUCKET>/staging" # GCS staging location for Dataflow
+      subnetwork                   = "<YOUR-FULL-PATH-SUBNETWORK"     # Give the full path to the subnetwork
+      temp_location                = "gs://<YOUR_GCS_BUCKET>/temp"    # GCS temp location for Dataflow
+      on_delete                    = "<YOUR_ON_DELETE_ACTION>"        # Action on Dataflow job deletion ("cancel" or "drain")
+      ip_configuration             = "<YOUR_IP_CONFIGURATION>"        # IP configuration for Dataflow workers ("WORKER_IP_PRIVATE" or "WORKER_IP_PUBLIC")
+    }
+  }
+}
+
+shard_list = [
+  {
+    shard_id = "<YOUR_SHARD_ID>" # A unique identifier for the shard (e.g., "shard-01")
```

Comment:
This structure is different from the sharding config created in the live migration flow. Will customers need to create this structure by hand ?

Reply:
> Will customers need to create this structure by hand ?

Yes, this will be the primary job they will have to do. They can write their own scripts which helps them generate the `.tfvars` file. For example, in my own 128 shard test, I wrote a python script which used a combination of querying the IP address of CloudSQL MySQL instances via gCloud and then writing out this structure for me.
The customer would have to do the same. BB could help, Gemini/ChatGPT would help here. Should be fairly straightforward.

> This structure is different from the sharding config created in the live migration flow.

Do you mean SMT? I am not using SMT as the benchmark for creating these configurations. I created this one trying to simplify as much as possible. If you look at `terraform_simple.tfvars`, the customer just needs to provide the connectivity details of there physical shards, that's it. This structure becomes significantly simpler than the SMT structure.

```
shard_list = [
  {
    dataflow_params = {
      mysql_host     = "<YOUR_MYSQL_HOST>"     # MySQL host address
      mysql_username = "<YOUR_MYSQL_USERNAME>" # MySQL username
      mysql_password = "<YOUR_MYSQL_PASSWORD>" # MySQL password
      mysql_port     = "<YOUR_MYSQL_PORT>"     # MySQL port (typically 3306)
    },
    "dataflow_params" : {
      "template_params" : {},
      "runner_params" : {}
    }
  }
]
```

Reply:
Let's chat about this offline. We need a version of that structure for other flows as well.

Reply:
Okay